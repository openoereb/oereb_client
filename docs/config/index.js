var path = require('canonical-path');

var Package = require('dgeni').Package;

module.exports = new Package('oereb_client_doc', [
  require('dgeni-packages/ngdoc'),
  require('dgeni-packages/nunjucks')
])

  .processor(require('./processors/index-page'))
  .processor(require('./processors/version'))
  .processor(require('./processors/api-data'))
  .processor(require('./processors/guide-data'))

  .config(function (log, readFilesProcessor, writeFilesProcessor) {

    // Set the log level to 'info', switch to 'debug' when troubleshooting
    log.level = 'info';

    // Specify the base path used when resolving relative paths to source and output files
    readFilesProcessor.basePath = path.resolve(__dirname, '../..');

    // Specify our source files that we want to extract
    readFilesProcessor.sourceFiles = [
      { // All of our application files
        include: 'oereb_client/static/js/**/*.js',
        basePath: 'oereb_client/static/js'
      },
      { // Our static Markdown documents
        include: 'docs/content/**/*.md',
        basePath: 'docs/content',
        fileReader: 'ngdocFileReader'
      }
    ];

    // Use the writeFilesProcessor to specify the output folder for the extracted files
    writeFilesProcessor.outputFolder = 'docs/build';

  })

  .config(function (parseTagsProcessor, getInjectables) {
    parseTagsProcessor.tagDefinitions =
      parseTagsProcessor.tagDefinitions.concat(getInjectables(require('./tag-defs')));
  })

  .config(function (templateFinder) {

    // Specify where the templates are located
    templateFinder.templateFolders.unshift(path.resolve(__dirname, 'templates'));

  })

  .config(function (computePathsProcessor, computeIdsProcessor) {

    computePathsProcessor.pathTemplates.push({
      docTypes: ['module'],
      pathTemplate: '${area}/${name}',
      outputPathTemplate: 'partials/${area}/${name}.html'
    });

    computePathsProcessor.pathTemplates.push({
      docTypes: ['componentGroup'],
      pathTemplate: '${area}/${moduleName}/${groupType}',
      outputPathTemplate: 'partials/${area}/${moduleName}/${groupType}.html'
    });

    // create new compute for 'content' type doc
    // indexPage is something new we will be defining later
    computeIdsProcessor.idTemplates.push({
      docTypes: ['content', 'indexPage'],
      getId: function (doc) {
        return doc.fileInfo.baseName;
      },
      getAliases: function (doc) {
        return [doc.id];
      }
    });

    // Build custom paths and set the outputPaths for "content" pages
    computePathsProcessor.pathTemplates.push({
      docTypes: ['content'],
      getPath: function (doc) {
        var docPath = path.dirname(doc.fileInfo.relativePath);
        if (doc.fileInfo.baseName !== 'index') {
          docPath = path.join(docPath, doc.fileInfo.baseName);
        }
        return docPath;
      },
      outputPathTemplate: 'partials/${path}.html'
    });

  })

  .config(function (extractAccessTransform) {
    extractAccessTransform.allowedDocTypes.add('js');
  });
