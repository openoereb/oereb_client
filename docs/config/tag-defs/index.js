module.exports = [
  require('./export'),
  require('./ngInject')
];