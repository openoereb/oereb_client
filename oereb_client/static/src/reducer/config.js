import {createSlice} from "@reduxjs/toolkit";

export const configSlice = createSlice({
  name: 'config',
  initialState: {
    config: {}
  },
  reducers: {
    update: (state, action) => {
      state.config = action.payload;
    }
  }
});

export const {update} = configSlice.actions;

export default configSlice.reducer;
