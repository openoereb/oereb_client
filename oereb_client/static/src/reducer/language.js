import {createSlice} from "@reduxjs/toolkit";

import i18n from "../i18n";

export const languageSlice = createSlice({
  name: 'language',
  initialState: {
    current: null,
    default: null,
    available: []
  },
  reducers: {
    initLanguages: (state, action) => {
      state.default = action.payload.default;
      state.available = action.payload.available;
      const query = new URLSearchParams(window.location.search);
      if (query.has('lang') && state.available.indexOf(query.get('lang')) > -1) {
        state.current = query.get('lang');
      }
      else {
        state.current = state.default;
      }
      i18n.changeLanguage(state.current);
      query.set('lang', state.current);
      window.history.pushState(null, null, '?' + query.toString());
    },
    setLanguage: (state, action) => {
      if (state.available.indexOf(action.payload) > -1) {
        state.current = action.payload;
        i18n.changeLanguage(state.current);
        const query = new URLSearchParams(window.location.search);
        query.set('lang', state.current);
        window.history.pushState(null, null, '?' + query.toString());
      }
    }
  }
});

export const {initLanguages, setLanguage} = languageSlice.actions;

export default languageSlice.reducer;
