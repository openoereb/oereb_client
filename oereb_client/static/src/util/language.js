import {isArray} from "lodash";

export const getLocalizedText = function (multilingualText, language, defaultLanguage) {
  if (isArray(multilingualText) && multilingualText.length > 0) {
    let defaultValue = null;
    for (let i = 0; i < multilingualText.length; i++) {
      if (multilingualText[i]["Language"] === language) {
        return multilingualText[i]["Text"];
      }
      else if (multilingualText[i]["Language"] === defaultLanguage) {
        defaultValue = multilingualText[i]["Text"];
      }
    }
    if (defaultValue) {
      return defaultValue;
    }
    return multilingualText[0]["Text"];
  }
  return null;
};

export const getLocalizedUrl = function (multilingualUrl, language, defaultLanguage) {
  if (isArray(multilingualUrl) && multilingualUrl.length > 0) {
    let defaultValue = null;
    for (let i = 0; i < multilingualUrl.length; i++) {
      if (multilingualUrl[i]["Language"] === language) {
        return multilingualUrl[i]["URL"];
      }
      else if (multilingualUrl[i]["Language"] === defaultLanguage) {
        defaultValue = multilingualUrl[i]["URL"];
      }
    }
    if (defaultValue) {
      return defaultValue;
    }
    return multilingualUrl[0]["URL"];
  }
  return null;
};
