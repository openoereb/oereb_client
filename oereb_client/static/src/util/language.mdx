import {Meta} from '@storybook/addon-docs/blocks';

<Meta title="API Reference/Util/Language" />

# getLocalizedText

Accepts a multilingual array containing localized texts for certain languages,
the requested language and the default language. The localized text of the
requested language is returned as result. If the requested language is not
available in the array, the default language is used as fallback.

```js
getLocalizedText(multilingualText, language, defaultLanguage);
```

### Parameters

| Param | Type | Description |
| --- | --- | --- |
| multilingualText | <code>Array</code> | The array containing the localized texts. |
| language | <code>String</code> | The requested language. |
| defaultLanguage | <code>String</code> | The default language used as fallback. |

### Returns

<code>String</code> The localized text.

# getLocalizedUrl

Accepts a multilingual array containing localized URLs for certain languages,
the requested language and the default language. The localized URL of the
requested language is returned as result. If the requested language is not
available in the array, the default language is used as fallback.

```js
getLocalizedUrl(multilingualUrl, language, defaultLanguage);
```

### Parameters

| Param | Type | Description |
| --- | --- | --- |
| multilingualUrl | <code>Array</code> | The array containing the localized URLs. |
| language | <code>String</code> | The requested language. |
| defaultLanguage | <code>String</code> | The default language used as fallback. |

### Returns

<code>String</code> The localized URL.
