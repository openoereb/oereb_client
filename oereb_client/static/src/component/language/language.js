import PropTypes from 'prop-types';
import React from "react";
import {withNamespaces} from 'react-i18next';
import {useDispatch, useSelector} from "react-redux";

import {setLanguage} from "../../reducer/language";

const OerebLanguage = function({t}) {
  const dispatch = useDispatch();
  const language = useSelector((state) => state.language);
  const currentLanguage = language.current;
  const availableLanguages = language.available;

  const selectLanguage = function(lang) {
    dispatch(setLanguage(lang));
  };

  const languages = availableLanguages.map((language, key) =>
    <li key={key}>
      <button className="dropdown-item" onClick={selectLanguage.bind(this, language)}>
        {language.toUpperCase()}
      </button>
    </li>
  );

  return (
    <div>
      <button className="btn btn-outline-secondary dropdown-toggle"
        type="button"
        data-bs-toggle="dropdown"
        aria-expanded="false"
        title={t('menu.language.title')}>
        {currentLanguage.toUpperCase()}
      </button>
      <ul className="dropdown-menu">
        {languages}
      </ul>
    </div>
  );
};

OerebLanguage.propTypes = {
  t: PropTypes.func.isRequired
};

export default withNamespaces()(OerebLanguage);
