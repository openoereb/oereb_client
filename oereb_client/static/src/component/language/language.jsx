import React from "react";
import {useTranslation} from "react-i18next";
import {useDispatch, useSelector} from "react-redux";

import {hideExtract, loadExtract, showError, showExtract} from "../../reducer/extract";
import {updateHistory} from "../../reducer/history";
import {setLanguage} from "../../reducer/language";
import {queryExtractById} from "../../request/extract";
import {TooManyRequestsError} from "../../util/error";
import {showWarning} from "../../reducer/message";

/**
 * This component provides the language selector.
 */
const OerebLanguage = function() {
  const {t} = useTranslation();
  const dispatch = useDispatch();
  const config = useSelector((state) => state.config).config;
  const serviceUrl = config.service_url;
  const language = useSelector((state) => state.language);
  const currentLanguage = language.current;
  const availableLanguages = language.available;
  const extract = useSelector((state) => state.extract);

  const selectLanguage = function(lang) {
    if (lang !== currentLanguage) {
      dispatch(setLanguage(lang));
      if (extract.visible) {
        dispatch(loadExtract({
          egrid: extract.egrid,
          identdn: extract.identdn,
          number: extract.number,
          zoom: false
        }));
        queryExtractById(
          serviceUrl,
          extract.egrid,
          extract.identdn,
          extract.number,
          config.extract_json_timeout,
          lang
        )
          .then((data) => {
            dispatch(showExtract(data));
            dispatch(updateHistory(data));
          })
          .catch((error) => {
            if (error instanceof TooManyRequestsError) {
              dispatch(hideExtract());
              dispatch(showWarning(t('extract.error.too_many_requests'), true));
            }
            else {
              dispatch(showError());
            }
          });
      }
    }
  };

  const languages = availableLanguages.map((language, key) =>
    <li key={key}>
      <button className="dropdown-item" onClick={selectLanguage.bind(this, language)}>
        {language.toUpperCase()}
      </button>
    </li>
  );

  return (
    <div className="oereb-client-language-selector">
      <button className="btn btn-outline-secondary dropdown-toggle"
        type="button"
        data-bs-toggle="dropdown"
        aria-expanded="false"
        title={t("menu.language.title")}>
        {currentLanguage.toUpperCase()}
      </button>
      <ul className="dropdown-menu">
        {languages}
      </ul>
    </div>
  );
};

export default OerebLanguage;
