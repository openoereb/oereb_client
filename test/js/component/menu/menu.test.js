import React from "react";
import {render, fireEvent, waitFor} from '@testing-library/react';
import {act} from "react-dom/test-utils";
import {Provider} from "react-redux";

import OerebMenu from "../../../../oereb_client/static/src/component/menu/menu";
import {update} from "../../../../oereb_client/static/src/reducer/config";
import {initLanguages} from "../../../../oereb_client/static/src/reducer/language";
import MainStore from "../../../../oereb_client/static/src/store/main";

const sleep = function(time) {
  return new Promise((resolve) => setTimeout(resolve, time));
};

const config = {
  application: {
    title: [
      {
        Language: 'en',
        Text: 'PLR Cadastre, Canton of Basel-Landschaft'
      },
      {
        Language: 'de',
        Text: 'ÖREB-Kataster, Kanton Basel-Landschaft'
      },
      {
        Language: 'fr',
        Text: 'Cadastre RDPPF, Canton de Bâle-Campagne'
      }
    ],
    logo_oereb: [
      {
        Language: 'en',
        URL: 'http://localhost:8080/samples/static/logo_oereb_en.png'
      },
      {
        Language: 'de',
        URL: 'http://localhost:8080/samples/static/logo_oereb_de.png'
      },
      {
        Language: 'fr',
        URL: 'http://localhost:8080/samples/static/logo_oereb_fr.png'
      }
    ],
    logo_canton: [
      {
        Language: 'en',
        URL: 'http://localhost:8080/samples/static/logo_bl.png'
      },
      {
        Language: 'de',
        URL: 'http://localhost:8080/samples/static/logo_bl.png'
      },
      {
        Language: 'fr',
        URL: 'http://localhost:8080/samples/static/logo_bl.png'
      }
    ],
    local_storage_prefix: 'bl',
    languages: ['en', 'de', 'fr'],
    default_language: 'en'
  },
  search_url: 'http://example.com/search'
};

describe('menu component', () => {

  let component;

  beforeEach(() => {
    act(() => {
      MainStore.dispatch(initLanguages({
        default: 'de',
        available: ['de']
      }));
    });
    act(() => {
      MainStore.dispatch(update(config));
    });
    component = render(
      <Provider store={MainStore}>
        <OerebMenu />
      </Provider>
    );
  });

  it('should render menu', () => {
    expect(component.asFragment()).toMatchSnapshot();
  });

});

describe('search', () => {

  let component;

  beforeEach(() => {
    fetch.resetMocks();
    act(() => {
      MainStore.dispatch(initLanguages({
        default: 'de',
        available: ['de']
      }));
    });
    act(() => {
      MainStore.dispatch(update(config));
    });
    component = render(
      <Provider store={MainStore}>
        <OerebMenu />
      </Provider>
    );
  });

  it('should show LV95 coordinates', async () => {
    fetch.mockResponseOnce('[]');
    fireEvent.change(component.container.querySelector('input'), {
      target: {
        value: '2600000 1200000'
      }
    });
    await sleep(500);
    await waitFor(() => {
      expect(component.asFragment()).toMatchSnapshot();
      expect(fetch.mock.calls).toHaveLength(1);
    });
  });

  it('should show GNSS coordinates', async () => {
    fetch.mockResponseOnce('[]');
    fireEvent.change(component.container.querySelector('input'), {
      target: {
        value: '7.72867 47.48910'
      }
    });
    await sleep(500);
    await waitFor(() => {
      expect(component.asFragment()).toMatchSnapshot();
      expect(fetch.mock.calls).toHaveLength(1);
    });
  });

  it('should show results', async () => {
    fetch.mockResponseOnce(JSON.stringify([
      {
        title: 'Set 1',
        results: [
          {
            label: 'Result 1',
            coordinates: [0.0, 0.0]
          },
          {
            label: 'Result 2',
            coordinates: [1.0, 1.0]
          }
        ]
      },
      {
        title: 'Set 2',
        results: [
          {
            label: 'Result 3',
            egrid: 'CH1234'
          },
          {
            label: 'Result 4',
            egrid: 'CH5678'
          }
        ]
      }
    ]));
    fireEvent.change(component.container.querySelector('input'), {
      target: {
        value: 'abc'
      }
    });
    await sleep(500);
    await waitFor(() => {
      expect(component.asFragment()).toMatchSnapshot();
      expect(fetch.mock.calls).toHaveLength(1);
    });
  });

});