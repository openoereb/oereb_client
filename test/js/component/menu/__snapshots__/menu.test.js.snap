// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menu component should render menu 1`] = `
<DocumentFragment>
  <div
    class="oereb-client-menu d-flex flex-column pb-3"
  >
    <div
      class="flex-grow-1 row justify-content-evenly"
    >
      <div
        class="col"
      >
        <img
          class="img-fluid p-2"
          src="http://localhost:8080/samples/static/logo_oereb_de.png"
        />
      </div>
      <div
        class="col text-end"
      >
        <img
          class="img-fluid p-2"
          src="http://localhost:8080/samples/static/logo_bl.png"
        />
      </div>
    </div>
    <div
      class="container-fluid"
    >
      <div
        class="input-group"
      >
        <button
          aria-expanded="false"
          class="btn btn-outline-secondary dropdown-toggle"
          data-bs-toggle="dropdown"
          title="menu.settings.title"
          type="button"
        >
          <i
            class="bi bi-gear-fill"
          />
        </button>
        <ul
          class="dropdown-menu"
        >
          <li>
            <button
              class="dropdown-item"
            >
              <span>
                <i
                  class="bi bi-check-square"
                />
                 menu.settings.availability
              </span>
            </button>
          </li>
          <li>
            <button
              class="dropdown-item"
            >
              <span>
                <i
                  class="bi bi-check-square"
                />
                 menu.settings.symbol_zoom
              </span>
            </button>
          </li>
        </ul>
        <button
          aria-expanded="false"
          class="btn btn-outline-secondary dropdown-toggle disabled"
          data-bs-toggle="dropdown"
          title="menu.history.title"
          type="button"
        >
          <i
            class="bi bi-clock-history"
          />
        </button>
        <ul
          class="dropdown-menu"
        />
        <input
          class="form-control"
          placeholder="menu.search.placeholder"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-secondary"
          disabled=""
          title="menu.search.reset"
          type="button"
        >
          <i
            class="bi bi-trash"
          />
        </button>
      </div>
    </div>
    <div
      class="container-fluid"
    />
  </div>
</DocumentFragment>
`;

exports[`search should show GNSS coordinates 1`] = `
<DocumentFragment>
  <div
    class="oereb-client-menu d-flex flex-column pb-3"
  >
    <div
      class="flex-grow-1 row justify-content-evenly"
    >
      <div
        class="col"
      >
        <img
          class="img-fluid p-2"
          src="http://localhost:8080/samples/static/logo_oereb_de.png"
        />
      </div>
      <div
        class="col text-end"
      >
        <img
          class="img-fluid p-2"
          src="http://localhost:8080/samples/static/logo_bl.png"
        />
      </div>
    </div>
    <div
      class="container-fluid"
    >
      <div
        class="input-group"
      >
        <button
          aria-expanded="false"
          class="btn btn-outline-secondary dropdown-toggle"
          data-bs-toggle="dropdown"
          title="menu.settings.title"
          type="button"
        >
          <i
            class="bi bi-gear-fill"
          />
        </button>
        <ul
          class="dropdown-menu"
        >
          <li>
            <button
              class="dropdown-item"
            >
              <span>
                <i
                  class="bi bi-check-square"
                />
                 menu.settings.availability
              </span>
            </button>
          </li>
          <li>
            <button
              class="dropdown-item"
            >
              <span>
                <i
                  class="bi bi-check-square"
                />
                 menu.settings.symbol_zoom
              </span>
            </button>
          </li>
        </ul>
        <button
          aria-expanded="false"
          class="btn btn-outline-secondary dropdown-toggle disabled"
          data-bs-toggle="dropdown"
          title="menu.history.title"
          type="button"
        >
          <i
            class="bi bi-clock-history"
          />
        </button>
        <ul
          class="dropdown-menu"
        />
        <input
          class="form-control"
          placeholder="menu.search.placeholder"
          type="text"
          value="7.72867 47.48910"
        />
        <button
          class="btn btn-outline-secondary"
          title="menu.search.reset"
          type="button"
        >
          <i
            class="bi bi-trash"
          />
        </button>
      </div>
    </div>
    <div
      class="container-fluid"
    />
  </div>
</DocumentFragment>
`;

exports[`search should show LV95 coordinates 1`] = `
<DocumentFragment>
  <div
    class="oereb-client-menu d-flex flex-column pb-3"
  >
    <div
      class="flex-grow-1 row justify-content-evenly"
    >
      <div
        class="col"
      >
        <img
          class="img-fluid p-2"
          src="http://localhost:8080/samples/static/logo_oereb_de.png"
        />
      </div>
      <div
        class="col text-end"
      >
        <img
          class="img-fluid p-2"
          src="http://localhost:8080/samples/static/logo_bl.png"
        />
      </div>
    </div>
    <div
      class="container-fluid"
    >
      <div
        class="input-group"
      >
        <button
          aria-expanded="false"
          class="btn btn-outline-secondary dropdown-toggle"
          data-bs-toggle="dropdown"
          title="menu.settings.title"
          type="button"
        >
          <i
            class="bi bi-gear-fill"
          />
        </button>
        <ul
          class="dropdown-menu"
        >
          <li>
            <button
              class="dropdown-item"
            >
              <span>
                <i
                  class="bi bi-check-square"
                />
                 menu.settings.availability
              </span>
            </button>
          </li>
          <li>
            <button
              class="dropdown-item"
            >
              <span>
                <i
                  class="bi bi-check-square"
                />
                 menu.settings.symbol_zoom
              </span>
            </button>
          </li>
        </ul>
        <button
          aria-expanded="false"
          class="btn btn-outline-secondary dropdown-toggle disabled"
          data-bs-toggle="dropdown"
          title="menu.history.title"
          type="button"
        >
          <i
            class="bi bi-clock-history"
          />
        </button>
        <ul
          class="dropdown-menu"
        />
        <input
          class="form-control"
          placeholder="menu.search.placeholder"
          type="text"
          value="2600000 1200000"
        />
        <button
          class="btn btn-outline-secondary"
          title="menu.search.reset"
          type="button"
        >
          <i
            class="bi bi-trash"
          />
        </button>
      </div>
    </div>
    <div
      class="container-fluid"
    >
      <div
        class="list-group result-list"
      >
        <div
          class="list-group-item"
        >
          <strong>
            menu.search.title.coordinates.lv95
          </strong>
          <span
            class="badge bg-secondary float-end"
          >
            1
          </span>
        </div>
        <button
          class="list-group-item list-group-item-action search-result text-start"
        >
          2600000 1200000
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`search should show results 1`] = `
<DocumentFragment>
  <div
    class="oereb-client-menu d-flex flex-column pb-3"
  >
    <div
      class="flex-grow-1 row justify-content-evenly"
    >
      <div
        class="col"
      >
        <img
          class="img-fluid p-2"
          src="http://localhost:8080/samples/static/logo_oereb_de.png"
        />
      </div>
      <div
        class="col text-end"
      >
        <img
          class="img-fluid p-2"
          src="http://localhost:8080/samples/static/logo_bl.png"
        />
      </div>
    </div>
    <div
      class="container-fluid"
    >
      <div
        class="input-group"
      >
        <button
          aria-expanded="false"
          class="btn btn-outline-secondary dropdown-toggle"
          data-bs-toggle="dropdown"
          title="menu.settings.title"
          type="button"
        >
          <i
            class="bi bi-gear-fill"
          />
        </button>
        <ul
          class="dropdown-menu"
        >
          <li>
            <button
              class="dropdown-item"
            >
              <span>
                <i
                  class="bi bi-check-square"
                />
                 menu.settings.availability
              </span>
            </button>
          </li>
          <li>
            <button
              class="dropdown-item"
            >
              <span>
                <i
                  class="bi bi-check-square"
                />
                 menu.settings.symbol_zoom
              </span>
            </button>
          </li>
        </ul>
        <button
          aria-expanded="false"
          class="btn btn-outline-secondary dropdown-toggle disabled"
          data-bs-toggle="dropdown"
          title="menu.history.title"
          type="button"
        >
          <i
            class="bi bi-clock-history"
          />
        </button>
        <ul
          class="dropdown-menu"
        />
        <input
          class="form-control"
          placeholder="menu.search.placeholder"
          type="text"
          value="abc"
        />
        <button
          class="btn btn-outline-secondary"
          title="menu.search.reset"
          type="button"
        >
          <i
            class="bi bi-trash"
          />
        </button>
      </div>
    </div>
    <div
      class="container-fluid"
    >
      <div
        class="list-group result-list"
      >
        <div
          class="list-group-item"
        >
          <strong>
            Set 1
          </strong>
          <span
            class="badge bg-secondary float-end"
          >
            2
          </span>
        </div>
        <button
          class="list-group-item list-group-item-action search-result text-start"
        >
          Result 1
        </button>
        <button
          class="list-group-item list-group-item-action search-result text-start"
        >
          Result 2
        </button>
      </div>
      <div
        class="list-group result-list"
      >
        <div
          class="list-group-item"
        >
          <strong>
            Set 2
          </strong>
          <span
            class="badge bg-secondary float-end"
          >
            2
          </span>
        </div>
        <button
          class="list-group-item list-group-item-action search-result text-start"
        >
          Result 3
        </button>
        <button
          class="list-group-item list-group-item-action search-result text-start"
        >
          Result 4
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
